ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -2009,12 +2006,15 @@ public float getProgress() throws IOException {
     final Reader.Options readOptions = OrcInputFormat.createOptionsForReader(conf);
     readOptions.range(split.getStart(), split.getLength());
 
-    String txnString = conf.get(ValidTxnList.VALID_TXNS_KEY);
-    ValidTxnList validTxnList = txnString == null ? new ValidReadTxnList() :
-      new ValidReadTxnList(txnString);
+    String txnString = conf.get(ValidWriteIdList.VALID_WRITEIDS_KEY);
+    ValidWriteIdList validWriteIdList
+            = (txnString == null) ? new ValidReaderWriteIdList() : new ValidReaderWriteIdList(txnString);
+    LOG.debug("getReader:: Read ValidWriteIdList: " + validWriteIdList.toString()
```

Comment:
Fixed.

Reply:
I'm not sure this fixes it.  what does "isAcidTable" in the Log message mean?  If it means "full acid" then checking isTableTransactional is not right - this will include MM tables.  So either "isAcidTable" should read "isTransactionalTable" or the value should be "isTableTransactional && (acidOperationalProperties == null or acidOperationalProperties.isSplitUpdate())"

Reply:
I think, this should be isTableTransactional as both ACID and MM need to have validWriteIdList.